Library/Homebrew/formula_installer.rb

```diff
@@ -752,6 +752,13 @@ def build
         sandbox.allow_write_xcode
         sandbox.allow_write_cellar(formula)
         sandbox.exec(*args)
+      elsif ARGV.no_sandbox? && ENV["HOMEBREW_NO_BUILD_EXEC"]
```

Comment:
Then we still call `exec` via the sandbox, but not explicitly in `FormulaInstaller` itself. That's probably misleading/unintuitive though, since `HOMEBREW_NO_BUILD_EXEC` implies that no `exec`s are done at all during a build.

I could make it a warning/failure condition, for clarity. Alternatively, I could remove the current fallthrough case and replace it with my no-`exec` case, which would allow us to remove `HOMEBREW_NO_BUILD_EXEC` entirely (it would just happen when `--no-sandbox` is passed).